Source/SwiftLintBuiltInRules/Rules/Lint/NonOptionalStringDataConversionRule.swift

```diff
@@ -1,31 +1,54 @@
 import SwiftSyntax
+import SwiftLintCore
+
+@AutoConfigParser
+struct NonOptionalStringDataConversionConfiguration: SeverityBasedRuleConfiguration {
+    typealias Parent = NonOptionalStringDataConversionRule
+
+    @ConfigurationElement(key: "severity")
+    private(set) var severityConfiguration = SeverityConfiguration<Parent>(.warning)
+    @ConfigurationElement(key: "include_variables")
+    private(set) var includeVariables = false
+}
 
 @SwiftSyntaxRule
 struct NonOptionalStringDataConversionRule: Rule {
-    var configuration = SeverityConfiguration<Self>(.warning)
+    var configuration = NonOptionalStringDataConversionConfiguration()
     static let description = RuleDescription(
         identifier: "non_optional_string_data_conversion",
         name: "Non-optional String -> Data Conversion",
         description: "Prefer non-optional `Data(_:)` initializer when converting `String` to `Data`",
         kind: .lint,
         nonTriggeringExamples: [
-            Example("Data(\"foo\".utf8)")
+            Example("Data(\"foo\".utf8)"),
+            Example("Data(string.utf8)", configuration: ["include_variables": true])
```

Comment:
Add a test that proofs that string literals still trigger with the option enabled.